### DISCUSSION TITLE
Autotype behavior

### DISCUSSION THREAD

**Discussion by josephj11** [keepassxreboot/keepassxc#6045] [Q&A]
I'm used to KeepassX. When I used it and more than one Autotype rule matched the current window, there was a simple pop-up that let me choose the correct entry and clicking on it used that entry.

KeepassXC appears a lot more complex than that. It switches to the KeepassXC window on a different virtual desktop and shows me the multiple entries, but doesn't apply them to the window where they're needed. It either does nothing or, if I select Perform Autotype, it does it in its own window. I have managed to get it to work several times, but never on the first try. (It works great if there's only one matching entry.)

What's the right way to do it when multiple entries match the current window?

I'm using KeepassXC 2.6.4 on kubuntu 18.08 Linux.

**Comment by droidmonkey** [comment]
This has been reported before and the cause is unknown why the behavior changed. It will probably be fixed with my autotype overhaul since I added code to explicitly refocus the desired window after choosing the autotype sequence.

**Reply by josephj11** [reply]
Great! Thanks. These systems that interact with big pieces of other people's code can be tough to debug. I support AutoKey which occasionally has quite similar issues.

**Reply by josephj11** [reply]
Is there a beta, etc. that I can clone and try out with a manageable expectation of safety (i.e. even if it doesn't work perfectly, it won't break my database or make it incompatible with the production version.) I can always copy my database first if it might get compromised and I make regular backups anyway.

**Reply by hifi** [reply]
Snapshots are available here: https://snapshot.keepassxc.org/latest/

We usually run the latest code ourselves so most of the time really nasty database breaking bugs are quickly fixed but it is indeed always a best practice to keep backups of your databases at all times.

**Reply by josephj11** [reply]
Thanks, I'll try out the AppImage which will leave my dpkg install intact (with backups!)
